src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Atomizer.java

```diff
@@ -0,0 +1,131 @@
+import al132.alchemistry.recipes.LiquifierRecipe;
+import al132.alchemistry.recipes.ModRecipes;
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.ingredient.IngredientHelper;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import org.jetbrains.annotations.Nullable;
+
+@RegistryDescription
+public class Atomizer extends VirtualizedRegistry<AtomizerRecipe> {
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(recipe -> ModRecipes.INSTANCE.getAtomizerRecipes().removeIf(r -> r == recipe));
+        ModRecipes.INSTANCE.getAtomizerRecipes().addAll(restoreFromBackup());
+    }
+
+    @RecipeBuilderDescription(example = {
+            @Example(".fluidInput(fluid('water') * 125).output(item('minecraft:clay'))"),
+            @Example(".fluidInput(fluid('lava') * 500).output(item('minecraft:gold_ingot')).reversible()")
+    })
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    public AtomizerRecipe add(AtomizerRecipe recipe) {
+        if (recipe != null) {
+            addScripted(recipe);
+            ModRecipes.INSTANCE.getAtomizerRecipes().add(recipe);
+        }
+        return recipe;
+    }
+
+    public boolean remove(AtomizerRecipe recipe) {
+        if (ModRecipes.INSTANCE.getAtomizerRecipes().removeIf(r -> r == recipe)) {
+            addBackup(recipe);
+            return true;
+        }
+        return false;
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByOutput", example = @Example(value = "item('alchemistry:compound:7')", commented = true))
+    public boolean removeByOutput(ItemStack output) {
+        return ModRecipes.INSTANCE.getAtomizerRecipes().removeIf(r -> {
+            if (r.getOutput().equals(output)) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByInput", example = @Example("fluid('water')"))
+    public boolean removeByInput(FluidStack input) {
+        return ModRecipes.INSTANCE.getAtomizerRecipes().removeIf(r -> {
+            if (r.getInput().isFluidEqual(input)) {
+                addBackup(r);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByInput")
+    public boolean removeByInput(IIngredient input) {
+        return removeByInput(IngredientHelper.toFluidStack(input));
+    }
```

Comment:
You can't just use `IIngredient` and then cast to `FluidStack` 💀

Reply:
:skull: